assignments/week_2/exercise_2_1/exercise_2_1.md

```diff
@@ -0,0 +1,173 @@
+# Exercise 2.1
+Welcome to the first exercise of week 2!
+
+# Project Objective
+Today we will be making a quick change to the code that we wrote for Project 1.3
+last week. We want to convert the code you wrote into a function so that we can call
+if multiple times without copy pasting code.
+
+Below is the solution the instructors wrote for this week. Feel free to work off of this
+directly or use your own code.
+
+```c++
+#include <iostream>
+#include <vector>
+
+std::vector<double> readInVector(std::string s) {
+  int prev_location = 0;
+  int next_location = 0;
+  std::vector<double> result;
+  while(s.find(',', prev_location) != std::string::npos) {
+    next_location = s.find(',', prev_location);
+    //std::cout << "prev_location: " << prev_location << std::endl;
+    //std::cout << "next_location: " << next_location << std::endl;
+    // substr [,]
+    //std::cout << s.substr(prev_location, next_location - prev_location) << std::endl;
+    result.push_back(std::stod(s.substr(prev_location, next_location - prev_location)));
+    next_location++;
+    prev_location = next_location;
+  }
+  result.push_back(std::stod(s.substr(prev_location, std::string::npos)));
+  return result;
+}
+
+int main() {
+  std::vector<double> x;
+  std::vector<double> w;
+  std::vector<double> y;
+  bool pack_with_zeros = true;
+
+  std::string s;
+  std::cin >> s;
+  if(s == "false") {
+    pack_with_zeros = false;
+  }
+  std::cin >> s;
+  x = readInVector(s);
+  std::cin >> s;
+  w = readInVector(s);
+
+
+  std::cout << "x: {" << x[0];
+  for(int i = 1; i < x.size(); i++) {
+    std::cout << ", " << x[i];
+  }
+  std::cout << "}" << std::endl;
+
+  std::cout << "w: {" << w[0];
+  for(int i = 1; i < w.size(); i++) {
+    std::cout << ", " << w[i];
+  }
+  std::cout << "}" << std::endl;
```

Comment:
Could we make them write two functions: one to do the convolution and one to print out a vector?